src/DenoHTTPWorker.ts

```diff
@@ -0,0 +1,377 @@
+import path, { resolve } from "path";
+import { ChildProcess, spawn, SpawnOptions, execSync } from "child_process";
+import { Readable, Writable, TransformCallback, Transform } from "stream";
+import readline from "readline";
+import http2 from "http2-wrapper";
+import got, { Got } from "got";
+
+import { fileURLToPath } from "url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const DENO_PORT_LOG_PREFIX = "deno-listening-port";
+const LISTENING_HOSTPORT = "0.0.0.0:0";
+
+const DEFAULT_DENO_BOOTSTRAP_SCRIPT_PATH = __dirname.endsWith("src")
+  ? resolve(__dirname, "../deno-bootstrap/index.ts")
+  : resolve(__dirname, "../../deno-bootstrap/index.ts");
+
+export interface DenoWorkerOptions {
+  /**
+   * The path to the executable that should be use when spawning the subprocess.
+   * Defaults to "deno". You can pass an array here if you want to invoke Deno
+   * with multiple arguments, like `sandbox run deno`.
+   */
+  denoExecutable: string | string[];
+
+  /**
+   * The path to the script that should be used to bootstrap the worker
+   * environment in Deno. If specified, this script will be used instead of the
+   * default bootstrap script. Only advanced users should set this.
+   */
+  denoBootstrapScriptPath: string;
+
+  /**
+   * Flags that are passed to the Deno process. These flags follow the `deno
+   * run` command and can be used to enabled/disabled various permissions and
+   * features. These flags will be modified to ensure that the deno process can
+   * run the http server we'll be connecting to. Toggle the
+   * printCommandAndArguments if you need to understand what the final flag
+   * values are.
+   *
+   * Review Deno's available flags here:
+   * https://docs.deno.com/runtime/manual/getting_started/command_line_interface
+   */
+  runFlags: string[];
+
+  /**
+   * Print stdout and stderr to the console with a "[deno]" prefix. This is
+   * useful for debugging.
+   */
+  printOutput: boolean;
+
+  /**
+   * Print out the command and arguments that are executed.
+   */
+  printCommandAndArguments: boolean;
+
+  /**
+   * Options used to spawn the Deno child process
+   */
+  spawnOptions: SpawnOptions;
+}
+
+export const newDenoHTTPWorker = async (
+  script: string | URL,
+  options?: Partial<DenoWorkerOptions>
+): Promise<DenoHTTPWorker> => {
+  const _options: DenoWorkerOptions = Object.assign(
+    {
+      denoExecutable: "deno",
+      denoBootstrapScriptPath: DEFAULT_DENO_BOOTSTRAP_SCRIPT_PATH,
+      runFlags: [],
+      printCommandAndArguments: false,
+      spawnOptions: {},
+      printOutput: false,
+    },
+    options || {}
+  );
```

Comment:
```suggestion
  const _options: DenoWorkerOptions = {
      denoExecutable: "deno",
      denoBootstrapScriptPath: DEFAULT_DENO_BOOTSTRAP_SCRIPT_PATH,
      runFlags: [],
      printCommandAndArguments: false,
      spawnOptions: {},
      printOutput: false,
      ...options
    };
```

Kind of JS fanciness but is more 'idiomatic' and produces a nicer inferred TypeScript type